File: 2016-10-30-CustomListViewSample/src/main/java/tech/thdev/customlistviewsample/view/presenter/MainContract.java
package tech.thdev.customlistviewsample.view.presenter;

import tech.thdev.customlistviewsample.data.GitHubItem;

/**
 * Created by tae-hwan on 10/25/16.
 */

public interface MainContract {

    interface View {

        void showProgress();

        void hideProgress();

        void notifyListView();

        void showFailLoad();

        void addItem(GitHubItem item);

        void showChromeTabs(GitHubItem item);

        GitHubItem getListItem(int position);
    }

    interface Presenter {

        void loadGitHubUser(String userKeyword);

        void onItemClick(int position);
    }
}

File: kotlin-example-01/src/main/java/tech/thdev/kotlin_example_01/view/main/presenter/MainContract.kt
package tech.thdev.kotlin_example_01.view.main.presenter

import tech.thdev.kotlin_example_01.base.presenter.BasePresenter
import tech.thdev.kotlin_example_01.base.presenter.BaseView
import tech.thdev.kotlin_example_01.view.main.adapter.model.PhotoDataModel

/**
 * Created by Tae-hwan on 7/21/16.
 */
interface MainContract {

    interface View: BaseView<Presenter> {

        fun showFailLoad()

        fun refresh()

        fun showProgress()

        fun hideProgress()

        fun initPhotoList()

        fun showBlurDialog(imageUrl: String?)

        fun showDetailView(imageUrl: String?)
    }

    interface Presenter: BasePresenter<View> {

        /**
         * Adapter setting.
         */
        fun setDataModel(model: PhotoDataModel)

        /**
         * Recent Photos
         */
        fun loadPhotos(page: Int)

        /**
         * keyword search photos
         */
        fun searchPhotos(page: Int, safeSearch: Int, text: String?)

        /**
         * Long click item
         */
        fun updateLongClickItem(position: Int)

        /**
         * Subscribe destroy.
         */
        fun unSubscribeSearch()

        /**
         * Load Detail Activity
         */
        fun loadDetailView(position: Int)
    }
}